### PR TITLE
Offir storagelite

### PR DESCRIPTION
**Pull request by offirko**
StorageLite file system will be introduced to mbed 5.9
I've submitted this pull request of StorageLite documentation and design for mbed's Handbook.
Thanks,
Offir

### CODE REVIEW

**Comment by AnotherButler** [comment]
Is this for porting, or is this for users?

**Comment by AnotherButler** [comment]
Also, is StorageLite on master yet? If not, could you please link to the open PR for the code? This lets me know if there's a code dependency I should watch for.

**Comment by dannybenor** [comment]
The design doc is here to help users to understand the feature, no need for a porting guide.
No PR for the code yet. Expect to be May 15

File: docs/reference/api/storage/StorageLite.md [hunk 1]
```diff
@@ -1,65 +1,73 @@
...
+- `file_exist`: verifies that file exists and authenticates it.
+- `get_file_flags`: returns files supported features (such as rollback protection, encryption and factory reset).
+- `get_first/next`: provides the ability to iterate through existing files, starting from the first one.
+- `factory_reset`: returns storage to contain only files that are backed up and returns those files to their backup version.
+
+##### StorageLiteFS
+
+- Mount or unmount: attaches or detaches a block device to or from the file system.
+- Open or close: establishes the connection between a file and a file descriptor.
+- Writing data to the file associated with the opened file descriptor (replacing existing data - if exists). You can only call write once.
```
Comment (offirko) [inline docs/reference/api/storage/StorageLite.md]: can you please add "'write' :" before the word writing

**Comment by AnotherButler** [comment]
Because it sounds more work may be done on this feature, I'm going to close this PR for now. Please feel free to reopen this later.

**Pull request closed without merging**